lib/puppet/resource/type_collection.rb

```diff
@@ -49,11 +49,6 @@ def inspect
     }.inspect
   end
 
-  def <<(thing)
```

Comment:
I think this should remain. We haven't had any deprecation warnings on it and it isn't @api private.